### ISSUE TITLE
workbox-webpack-plugin - `chunks` option in conflict with precaching `index.html`

### ISSUE THREAD

**Issue by pedi** [GoogleChrome/workbox#1556]
**Library Affected**:
*workbox-webpack-plugin*

**Browser & Platform**:
*all browsers*

**Issue or Feature Request Description**:
Background here: we are using the `webpack-html-plugin` to generate `index.html` and `workbox-webpack-plugin` to inject the `__precacheManifest` into our service worker file.

When `chunks` option is not specified, `InjecInjectManifest`will generate a list of all the files compiled by `webpack` (including all the static assets like html, images etc). However this is not really an option for us, as we have 100+ JS files. It's not wise to precache all of them upfront.

So we are using the `chunks` option to only precache the application shell related code. However, since the `webpack-html-plugin` generated html will have no associated chunks. The `index.html` will be filtered out from the `precacheManifest`

I tried to work around the issue by using the old `glob` way. However it's meant only for external files, not the webpack compilations. 

I dig into the source code and seems no way to work around this issue. I'm pretty sure this is a common use case for large websites like us, am I missing anything here or this is a feature by     ####design?

**Comment by jeffposnick** [comment]
Hello @pedi—Sorry that you're running into that. As an alternative to `chunks`, you could potentially make use of one of the following [supported options](https://developers.google.com/web/tools/workbox/modules/workbox-webpack-plugin#full_generatesw_config):

- `excludeChunks`, if it's cleaner to list a set of chunk names to avoid caching. Assets that don't match, like your `index.html`, should still be precached.

- `include` and `exclude`, to specify one or more RegExp patterns that will match against asset names and allow you to opt-in or opt-out of precaching certain assets.

Do either of those options sound like they'd help?

**Comment by pedi** [comment]
Thanks @jeffposnick for quick follow up on this.

Option 1 might not be feasible for us as the `exclude` list will be long and hard to maintain

Option 2 can serve as a workaround for us at the moment. However the chunks still provide a lower long -term maintenance cost as more files could be emitted from the same chunk in future and their file names might not necessarily match the predefined regex.

Currently our workaround is to to compile `sw.js` as a separate webpack stage and use the built one as a template to feed into the `InjectManifest` plugin. 

Hope the code snippet below can explain our approach clearly.

```
const precacheFileList = self.__precacheManifest;

workbox.precaching.precacheAndRoute(
  precacheFileList.concat([
    {
      url: 'index.html',
      revision: __GIT_SHA__,
    },
  ])
);
```

**Comment by jeffposnick** [comment]
This should be addressed by the current Workbox v5.0.0 alpha.
